Source/IntegrationTests/TestFiles/LitTests/LitTest/ast/assume.dfy

```diff
@@ -1,6 +1,6 @@
-// RUN: %testDafnyForEachResolver "%s" -- --allow-axioms=false 
+// RUN: ! %testDafnyForEachResolver "%s" -- --allow-axioms=false
```

Comment:
Ooh this strikes me as too dangerous, because it could hide a complete crash in `%testDafnyForEachResolver`. 

`!` as a littish command is fairly dangerous in general, because we tend to only forward stdout to `%t` and not stderr, which means tests don't fail if the output matches but there's some unrelated fatal error (and I've seen that happen: https://github.com/dafny-lang/dafny/pull/5152)

Is it a lot more effort to add `--allow-warnings` everywhere instead? Probably a bit of work to ensure `%testDafnyForEachResolver` supports it, but I think that's worth it.

I actually wonder if we should make our own `NotCommand` implementation of `!` fail if stderr receives any output, or perhaps even ANY command. That would no longer be consistent with `lit`, but it might be worth it.

Reply:
> Ooh this strikes me as too dangerous

That is not the general approach I have taken to fixing the tests, but for this test in particular, I want to assert that the exit code is not success.

Reply:
Sure, if you want to assert that testDafnyForEachCompiler behaves correctly when there are warnings and --allow-warnings is passed, add that case to the metatests folder. I just don't want to add ! to existing tests, because that weakens their coverage.

Reply:
I think testing for a success or failure exit code does not necessarily say anything about the stdout and stderr output, so you need to test those separately. If you do that, then there is no difference in safety between testing for success or failure exit codes.

I've recently added operators to our lit support to make tracking both stdout and stderr easy, using `&>` and `&>>` (https://github.com/dafny-lang/dafny/blob/master/Source/XUnitExtensions/Lit/LitCommandWithRedirection.cs#L56).

I'll update all tests where I added `!` so they uses `&>` instead of `>`. Does that alleviate your concerns?

Reply:
YES, love it! Great idea. As a follow up we should be using `&>` and `&>>` in every single littish test to be safe, unless the test specifically wants to handle stdout and stderr separately.